src/marqo/tensor_search/models/index_settings.py

```diff
@@ -45,15 +45,19 @@ class IndexSettings(StrictBaseModel):
     @root_validator(pre=True)
     def validate_field_names(cls, values):
         # Verify no snake case field names (pydantic won't catch these due to allow_population_by_field_name = True)
-        def validate_dict_keys(d: dict):
-            for key in d.keys():
-                if '_' in key:
-                    raise ValueError(f"Invalid field name '{key}'. "
-                                     f"See Create Index API reference here https://docs.marqo.ai/2.0.0/API-Reference/Indexes/create_index/")
-                if isinstance(d[key], dict):
-                    validate_dict_keys(d[key])
-
-        validate_dict_keys(values)
+        def validate_keys(d: Union[dict, list]):
+            if isinstance(d, dict):
+                for key in d.keys():
+                    if '_' in key:
+                        raise ValueError(f"Invalid field name '{key}'. "
```

Comment:
Should we explicitly specify that the error is related to `snake_case` or `camelCase` in IndexSettings field name here to avoid confusion? E.g.,there is also a `name` in `allFields`

Reply:
I thought that might be too much information and confusing. Ideally our pydantic model should error out for these fields like any other invalid field name. The reason we need this hack is our use of aliases. I think this difference in implementation shouldn't affect the user experience and error message.